File: kelmah-mobile-android/app/src/main/java/com/kelmah/mobile/core/session/KelmahUserRole.kt
package com.kelmah.mobile.core.session

import com.kelmah.mobile.core.storage.SessionUser

enum class KelmahUserRole {
    WORKER,
    HIRER;

    val title: String
        get() = if (this == HIRER) "Hirer" else "Worker"

    companion object {
        fun from(role: String?): KelmahUserRole =
            if (role.equals("hirer", ignoreCase = true)) HIRER else WORKER
    }
}

val SessionUser?.kelmahUserRole: KelmahUserRole
    get() = KelmahUserRole.from(this?.role)

File: kelmah-mobile-ios/Kelmah/Core/Session/KelmahUserRole.swift
import Foundation

enum KelmahUserRole {
    case worker
    case hirer

    init(role: String?) {
        if role?.lowercased() == "hirer" {
            self = .hirer
        } else {
            self = .worker
        }
    }

    var title: String {
        switch self {
        case .worker:
            return "Worker"
        case .hirer:
            return "Hirer"
        }
    }
}

extension SessionUser {
    var kelmahUserRole: KelmahUserRole {
        KelmahUserRole(role: role)
    }
}